WooCommerce/Classes/ViewRelated/Orders/Cells/OrderTableViewCell.swift

```diff
@@ -147,4 +149,17 @@ private extension OrderTableViewCell {
 
         dateCreatedLabel.applyCaption1Style()
     }
+
+    // Assure top and bottom borders when cell is highlighted
+    func configureBordersWhenHighlighted() {
+        let topBorder = CALayer()
+        let bottomBorder = CALayer()
+        let borderWidth = 0.5
+
+        topBorder.frame = CGRect(x: 0, y: 0, width: frame.width, height: borderWidth)
+        bottomBorder.frame = CGRect(x: 0, y: 0, width: frame.width, height: borderWidth)
+        topBorder.backgroundColor = UIColor.divider.cgColor
+        layer.addSublayer(topBorder)
+        layer.addSublayer(bottomBorder)
+    }
```

Comment:
> The fix should probably be in UITableViewCell.updateDefaultBackgroundConfiguration(using:style:) so we get it everywhere...

Yeah. I experimented with different approaches and the simplest one seems to be using the `.stroke...` properties inside the cell configuration. Here's the change: https://github.com/woocommerce/woocommerce-ios/pull/12214/commits/ffed3e920a4ea83949b52fb1ccd6cc342d4a3d03

I wasn't able to find a good way to apply this change only to top and bottom: 
* If we use CALayer we have to also play with removing them after a cell stops being selected/highlighted, which I wasn't able to do without crashing when populating the cells. This also adds additional unexpected lines around the first order and we'll have to add additional handling most likely by overriding `layoutSubviews`
* Using the [customView](https://developer.apple.com/documentation/uikit/uibackgroundconfiguration/3600322-customview) property feels a bit overkill to add these borders, at least for the moment. If we happen to redo the cells across the app this could make more sense.

The downside of the approach is that we can see the lateral border a bit wider when a cell is selected, but I think it makes sense and overall is an improvement:
![simulator_screenshot_39C4B052-902D-4FA2-B141-D366C88AF3E6](https://github.com/woocommerce/woocommerce-ios/assets/3812076/b14f886b-9d7b-48bd-9a78-09465a8f8434)